From 27b86bc2dd485fadf37b5f4cf2b553e5ea7f1617 Mon Sep 17 00:00:00 2001
From: Wan Ahmad Zainie <wan.ahmad.zainie.wan.mohamad@intel.com>
Date: Fri, 21 Sep 2018 16:33:52 +0800
Subject: [PATCH] usb: roles: intel_xhci: Add default_role module parameter

This patch adds default_role module parameter for
intel_xhci_usb_role_switch.ko. This will allow people to override
default role set by BIOS when XDCI Support is set to PCI Mode.

Signed-off-by: Wan Ahmad Zainie <wan.ahmad.zainie.wan.mohamad@intel.com>
---
 drivers/usb/roles/intel-xhci-usb-role-switch.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/drivers/usb/roles/intel-xhci-usb-role-switch.c b/drivers/usb/roles/intel-xhci-usb-role-switch.c
index f9ed262..f09bcc1 100644
--- a/drivers/usb/roles/intel-xhci-usb-role-switch.c
+++ b/drivers/usb/roles/intel-xhci-usb-role-switch.c
@@ -44,6 +44,10 @@ struct intel_xhci_usb_data {
 	bool disable_sw_switch;
 };
 
+static int default_role;
+module_param(default_role, int, 0444);
+MODULE_PARM_DESC(default_role, "USB OTG port default role 0:default 1:host 2:device");
+
 static int intel_xhci_usb_set_role(struct device *dev, enum usb_role role)
 {
 	struct intel_xhci_usb_data *data = dev_get_drvdata(dev);
@@ -178,6 +182,16 @@ static int intel_xhci_usb_probe(struct platform_device *pdev)
 	pm_runtime_set_active(dev);
 	pm_runtime_enable(dev);
 
+	/* Override USB OTG port default role. */
+	switch (default_role) {
+	case USB_ROLE_HOST:
+		intel_xhci_usb_set_role(dev, USB_ROLE_HOST);
+		break;
+	case USB_ROLE_DEVICE:
+		intel_xhci_usb_set_role(dev, USB_ROLE_DEVICE);
+		break;
+	}
+
 	return 0;
 }
 
-- 
1.9.1

